cpp/src/rolling/detail/rolling_utils.cu

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "rolling.cuh"
+#include "rolling.hpp"
+
+#include <cudf/aggregation.hpp>
+#include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+namespace cudf {
+namespace detail {
+namespace {
+struct is_supported_rolling_aggregation_impl {
+  template <typename T, aggregation::Kind kind>
+  constexpr bool operator()() const noexcept
+  {
+    return (kind == aggregation::Kind::LEAD || kind == aggregation::Kind::LAG ||
+            kind == aggregation::Kind::COLLECT_LIST || aggregation::Kind::COLLECT_SET) ||
+           corresponding_rolling_operator<T, kind>::type::is_supported();
```

Comment:
`corresponding_rolling_operator` is defined in a `.cuh` header so I think without bigger refactoring I need this to be compiled in a TU with nvcc (even though we don't need any device-side code).

Reply:
thanks, I'll give that a go

Reply:
Ok. Perhaps the `cuda_runtime.h` is missing somewhere. Including after all the other includes would not help.

Why is the `rolling.cuh` needed to be included in this file?

Reply:
> Why is the rolling.cuh needed to be included in this file?

This dispatch to find out if a rolling aggregation is supported uses the constexpr static `is_supported()` method of the `corresponding_rolling_operator` struct that is defined in `rolling.cuh`.

Reply:
That's too bad. It looks everything it needs does not require device code. The `is_supported()` there could be moved into a .hpp header without an issue. But I think it would be out of scope for this PR.

Reply:
Can we open an issue with the suggested change to track as future work?

Reply:
Done.